### ISSUE TITLE
"Client does not support authentication protocol requested by server; consider upgrading MySQL client (Exception)"

### ISSUE THREAD

**Issue by FredTheHomie** [crystal-lang/crystal-mysql#56]
I get the below error when I try to run the example test code. I am not sure what is causing this. Any help would be greatly appreciated. Just as a note, I have the latest version of MySql.
[
<img width="1079" alt="screen shot 2018-05-30 at 00 41 04" src="https://user-images.githubusercontent.com/20875271/40691099-3adeff42-63a2-11e8-954b-75f777219656.png">
](url)

**Comment by bcardiff** [comment]
@FredTheHomie specs are run agains 5.6 and 5.7 only. Probably a bit old if you are running latest version. 

Would you mind telling the exact version you are using?
If you like binary protocol, or are willing to, feel free to send a PR implementing part of the protocol that might have changed from the above version to yours.

**Comment by rdp** [comment]
I ran into this, worked around it by setting a "normal" password for root, FWIW (see comments here: https://dev.mysql.com/doc/refman/5.5/en/old-client.html )

**Comment by mamantoha** [comment]
I have the same issue with MySQL 8.0.13.

**Comment by fernandes** [comment]
yeah, it's because the README code initialize without any password for user `root`, setting a normal user / password, or password for root works flawlessly

**Comment by fernandes** [comment]
with mysql there's a problem with packet 254, probably related to #62 , trying to figure out

**Comment by rdp** [comment]
I don't think this was exactly fixed (though a work around was mentioned in the README)...consider reopen'ing?
